src/main/java/Sorting.java

```diff
@@ -0,0 +1,273 @@
+
+
+import java.util.*;
+
+/**
+ * Created by sam on 6/17/16.
+ */
+public class Sorting {
+
+    // implement bubble sort
+    /*
+        input heapArray of sortable items?
+         ints for now
+         while unsorted
+          loop thru entire heapArray and if an element is less than the previous element then swap them
+          if looped thru the entire heapArray and every element is >= previous, sorted
+     */
+
+    public static List<Integer> bubbleSort(List<Integer> array) throws IllegalArgumentException {
+
+        if(array == null) {
+            throw new IllegalArgumentException("invalid input array to bubble sort: input array is null");
+        }
+
+        if(array.isEmpty()) {
+            throw new IllegalArgumentException("invalid input array to bubble sort: empty array");
+        }
+
+        boolean unsorted = true;
+        while(unsorted) {
+            unsorted = false;
+            for(int i = 1; i < array.size(); i++) {
+                if(array.get(i) < array.get(i - 1)) {
+                    Collections.swap(array, i, i -1);
+                    unsorted = true;
+                }
+            }
+        }
+
+        return array;
+    }
+
+    // implement merge sort
+    /*
+        input heapArray of sortable items?
+         ints for now
+         cut heapArray in half recursively until each heapArray is size 1
+         merge the two subarrays by grabbing the next from the two arrays that fits sort condition
+         for sorting ints ascending, grab lower of two ints
+     */
+    public static List<Integer> mergeSort(List<Integer> array)  throws IllegalArgumentException {
+        if(array == null) {
+            throw new IllegalArgumentException("invalid input array to merge sort: input array is null");
+        }
+
+        if(array.isEmpty()) {
+            throw new IllegalArgumentException("invalid input array to merge sort: empty array");
+        }
+
+        return mergeSort(array, 0, array.size() - 1);
+    }
+
+    public static List<Integer> mergeSort(List<Integer> array, int start, int end) throws IllegalArgumentException {
+
+        if(start < end) {
+            int middle = (int)Math.floor((start + end) / 2d);
+            mergeSort(array, start, middle);
+            mergeSort(array, middle + 1, end);
+            merge(array,start,middle,end);
+        }
+
+        return  array;
```

Comment:
You're operating on my input! I'd suggest copying 'array' into a new List (and renaming 'array' to something like toBeSorted, because its a List). That way, I can get back a sorted List, and still have my original input (which maybe I didn't want mangled if I need to do other things with it!).